### ISSUE TITLE
Redundant imports under the same package haven't been removed

### ISSUE THREAD

**Issue by Goooler** [pinterest/ktlint#3004]
## Expected Behavior

[`Foo.kt`](https://github.com/user-attachments/files/20620452/Foo.kt.txt):

```kt
package com.example

object Foo
```

[`Bar.kt`](https://github.com/user-attachments/files/20620471/Bar.kt.txt):

```kt
package com.example

import com.example.Foo

fun bar(): Foo = Foo
```

The import of `com.example.Foo` in `Bar.kt` is redundant, it should be removed by Ktlint formatting.

## Observed Behavior

Run Ktlint check:

```sh
ktlint . # Nothing wrong with this example.
```

Run Ktlint format:

```sh
ktlint -F . # Nothing changed with this example.
```

## Steps to Reproduce

The steps above.

## Your Environment

* Version of ktlint used: CLI `1.6.0`

**Comment by Goooler** [comment]
May something be related to https://github.com/pinterest/ktlint/issues/1754#issuecomment-1368201667?

**Comment by paul-dingemans** [comment]
> May something be related to [#1754 (comment)](https://github.com/pinterest/ktlint/issues/1754#issuecomment-1368201667)?

Yes definitely. I am inclined to remove the import rule entirely. It does not really fit in the concept of a linter/formatter aiming to prettify the code. It is more fit for statical code analyzer that does more in depth checks on code.

**Comment by Goooler** [comment]
Thanks for the explanation!
